tests/foreman/ui/test_contentview.py

```diff
@@ -847,11 +846,42 @@ def test_negative_add_dupe_modules(self):
 
         @assert: User cannot add modules multiple times to the view
 
-        @caseautomation: notautomated
-
-
         @CaseLevel: Integration
         """
+        cv_name = gen_string('alpha')
+        module_name = 'samba'
+        product = entities.Product(organization=self.organization).create()
+        puppet_repository = entities.Repository(
+            url=FAKE_0_PUPPET_REPO,
+            content_type='puppet',
+            product=product
+        ).create()
+        puppet_repository.sync()
+        with Session(self.browser) as session:
+            # Create content-view
+            make_contentview(session, org=self.organization.name, name=cv_name)
+            self.assertIsNotNone(self.content_views.search(cv_name))
+            self.content_views.add_puppet_module(
+                cv_name,
+                module_name,
+                filter_term='Latest'
+            )
+            # ensure that the puppet module is added to content view
+            self.assertIsNotNone(
+                self.content_views.fetch_puppet_module(
+                    cv_name, module_name)
+            )
+            # ensure that cannot add the same module a second time.
+            with self.assertRaises(UINoSuchElementError) as context:
+                self.content_views.add_puppet_module(
+                    cv_name,
+                    module_name,
+                    filter_term='Latest'
+                )
+                # ensure that the select location of our module is in the
+                # exception message
+                _, location = locators.contentviews.select_module % module_name
+                self.assertIn(location, context.exception.message)
```

Comment:
self.assertRaises() executes the code from block, catches all the exceptions and checks whether the passed one was risen. So in case of few exceptions risen only one will actually be checked. That makes `self.assertIn()` check pointless, as in case it fails but `UINoSuchElementError` was risen - the test will pass.
Thus it's not just the style, the logic will be wrong if we don't put the code which may rise different exception outside of the `with self.assertRaises(...` block.